crates/vim/src/normal/increment.rs

```diff
@@ -120,10 +123,12 @@ fn increment_decimal_string(mut num: &str, mut delta: i64) -> String {
         u64::MAX
     };
 
-    if result == 0 || !negative {
-        format!("{}", result)
+    if result == 0 {
+        '0'.to_string()
```

Comment:
@ThomAub I think this should still be padded to width?

Reply:
After a double check in vim/neovim 
`1ˇ2` `12 <C-x>`  becomes `0` and stay like this 
One other broken edge case is 
`09ˇ9` `<C-a> <C-x>` is `99` with vim not `099`

Reply:
Hmm; but `012` becomes `000`.

I think we need to look at the first digit to decide whether to preserve the width? e.g. 100 ctrl-x should become 99 (not 099).

Reply:
Indeed, I mostly use `<C-a>` to reflow some migrations number and never really looked at all the possibilities of 
`<C-a> <C-x>`.
I will study how [neovim does this](https://github.com/neovim/neovim/blob/64847fbdc908bf0a301b8f1e1814ff71bd425bae/src/nvim/ops.c#L4510) and more tests case in following commits